subprojects/core/src/main/java/org/gradle/execution/plan/ExecutionDependencies.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.plan;
+
+import org.gradle.api.Task;
+
+import java.util.Set;
+
+public interface ExecutionDependencies {
```

Comment:
I was wondering if we could push the visitor pattern even further out into this interface. So you would ask to visit the dependencies of a thing, and then you could decide what you care about - tasks or transformations or both.

Do you think that would simplify the code?

Reply:
I tried that but it would rather complicate the code because all visitors currently want to collect items and thus the code for doing do would end up being duplicated.